Rpi eeprom

LostFuse opened the pull request:
- What I did
Added rpi-eeprom recipe to provide bootloader update

- How I did it

    Added rpi-eeprom recipe
    Disabled VC4 for graphics to fall back using userland

Commit:
machine/raspberrypi4-64.conf: disabled vc4 for graphicB

Signed-off-by: leckijakub <jakub.lecki@3mdeb.com>

Commit:
rpi-eeprom: new recipe

Signed-off-by: leckijakub <jakub.lecki@3mdeb.com>

Commit:
rpi-eeprom: deleted unnecessary instructions

Signed-off-by: leckijakub <jakub.lecki@3mdeb.com>

Commit:
rpi-eeprom: deleted deploy inheritance

Signed-off-by: leckijakub <jakub.lecki@3mdeb.com>

Commit:
rpi-eeprom: deleted deploy task

Signed-off-by: leckijakub <jakub.lecki@3mdeb.com>

Commit:
rpi-eeprom: deleted unused default-config.txt

Signed-off-by: leckijakub <jakub.lecki@3mdeb.com>

macpijan commented on conf/machine/raspberrypi4-64.conf:
```diff
@@ -34,3 +34,6 @@ KERNEL_BOOTCMD ?= "booti"
 ARMSTUB ?= "armstub8-gic.bin"
+
+# disable vc4 for graphics to fall back using userland (dagerous!)
+DISABLE_VC4GRAPHICS = "1"
```
@leckijakub Why **exactly** is it necessary? Do we have some error messages if it was not set?

macpijan commented on recipes-bsp/rpi-eeprom/rpi-eeprom_git.bb:
```diff
@@ -0,0 +1,67 @@
+FILESEXTRAPATHS_append := ":${THISDIR}/files"
```
This should not be needed

macpijan commented on recipes-bsp/rpi-eeprom/rpi-eeprom_git.bb:
```diff
@@ -0,0 +1,67 @@
+COMPATIBLE_MACHINE = "raspberrypi4-64"
+
+DEPENDS += "python3-native"
+RDEPENDS_${PN} += "coreutils python3 userland"
```
Which components from this rpi-eeprom need the userland? Are they crucial for performing the eeprom firmware update?

macpijan committed:
userland: add userlandtools recipe

The goal is that the userlandtools recipe contains some tools coming
from the userland, which are required to perform the EEPROM update. At
the same time, we do not want to use the full userland, as we prefer to
use the mainline mesa userspace GPU libraries.

As part of this work, userland.inc file was introduced. It contains
common parts for the userland and userlandtools recipes.

The userlandtools recipe is inspired by the one from Balena:
https://github.com/balena-os/balena-raspberrypi/blob/master/layers/\
  meta-balena-raspberrypi/recipes-extended/userlandtools/userlandtools.bb

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
Revert "machine/raspberrypi4-64.conf: disabled vc4 for graphicB"

This reverts commit c3173f262f303f5e0e15c792b1a9ccc3d97d48d6.

macpijan committed:
rpi-eeprom: fix recipe

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>
